pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,319 @@
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+"""
+Implementation of the BaseRepository that implements an in-memory repository
+based on dictionaries
+
+For the module-level documentation, see mocksupport.rst.
+"""
+
+import six
+
+from pywbem._nocasedict import NocaseDict
+
+from pywbem import CIMClass, CIMQualifierDeclaration, CIMInstance
+from pywbem._utils import _format
+from ._baserepository import BaseObjStore, BaseRepository
+
+
+class InMemoryObjStore(BaseObjStore):
+    """
+    """
+    def __init__(self, case_insensitive_names, cim_object_type, repository):
+        """
+
+          Parameters:
+
+            case_insensitive_names(:class:`py:bool`):
+                Defines whether the data store for this type handles
+                case insesitive names or not
+
+            cim_object_type (TODO):
+             CIMType of the object storded
+
+            repository (TODO)
+                handle of the repository for the type defined in cim_object_Type
+                within a single namespace
+        """
+
+        # TODO namespace must point to dynamic object that changes when
+        # namespaces added.
+        self._case_insensitive_names = case_insensitive_names
+        self._cim_object_type = cim_object_type
+
+        self._data = NocaseDict() if case_insensitive_names else {}
+
+    def __repr__(self):
+        return _format('InMemoryObjStore(type={0},  dict={1}, size {2}',
+                       self._cim_object_type, type(self._data),
+                       len(self._data))
+
+    def exists(self, name):
+        """
+        Test if the object defined by name and namespace exists in the object
+        store
+        """
+        return True if name in self._data else False
+
+    def get(self, name, copy=True):
+        """
+        """
+        #print("GET %s" % name)
+        if name in self._data:
+            if copy:
+                return self._data[name].copy()
+            else:
+                return self._data[name]
+        else:
+            raise KeyError('{} not in datastore'.format(name))
+
+    def create(self, name, cim_object):
+        """
+        """
+        #print("CREATE %s %s " % (name, self._cim_object_type))
+        assert isinstance(cim_object, self._cim_object_type)
+        if name in self._data:
+            #print('CREATE already in datastoe')
+            raise ValueError('{} already in datastore'.format(name))
+        else:
+            self._data[name] = cim_object
```

Comment:
So it does not do a copy in this code. Please adjust the code to whatever the intention was of the base class.